src/arrayMethodReduce.js

```diff
@@ -5,7 +5,17 @@
  */
 function applyCustomReduce() {
   [].__proto__.reduce2 = function(callback, initialValue) {
-    // write code here
+    let value = arguments.length > 1
+      ? initialValue : this[0];
+
+    const iterations = arguments.length > 1
+      ? 0 : 1;
+
+    for (let i = iterations; i < this.length; i++) {
+      value = callback(value, this[i], i, this);
+    }
```

Comment:
We on `Array iteration methods` topic. So maybe should use some array method instead of a simple loop )

Reply:
I am not sure if I can use array methods in this task, because on MA platform all tasks, where custom method has to be implemented, standard array methods are forbidde
<img width="320" alt="Снимок экрана 2022-08-17 в 23 53 13" src="https://user-images.githubusercontent.com/107410035/185240960-fe510486-5565-4934-a7c0-764a530e280c.png">
n